libs/src/litmus_libs/interfaces/litmus_auth.py

```diff
@@ -0,0 +1,225 @@
+
+
+class AuthRequirerAppDatabagModel(Endpoint):
+    """Requirer application databag model for the litmus_auth interface."""
+
+    version: int = 0
+
+
+class LitmusAuthDataProvider:
+    """Wraps a litmus_auth provider endpoint."""
+
+    def __init__(
+        self,
+        relations: List[ops.Relation],
+        app: ops.Application,
+        secret_setter: Callable[..., ops.Secret],
+        secret_getter: Callable[..., ops.Secret],
+    ):
+        self._relations = relations
+        self._app = app
+        self._secret_setter = secret_setter
+        self._secret_getter = secret_getter
+
+    def _store_dex_config_in_secret(self, dex: DexConfig) -> Optional[ops.Secret]:
+        if dex is None:
+            return None
+        content = {}
+        if dex.dex_oauth_client_secret is not None:
+            content["dex-oauth-client-secret"] = dex.dex_oauth_client_secret
+        if dex.oauth_jwt_secret is not None:
+            content["oauth-jwt-secret"] = dex.oauth_jwt_secret
+        if not content:
+            return None
+        try:
+            secret = self._secret_getter(label=DEX_SECRET_LABEL)
+            secret.set_content(content)  # pyright: ignore
+            secret.get_content(refresh=True)
+            return secret
+        except ops.SecretNotFoundError:
+            secret = self._secret_setter(
+                content=content,
+                label=DEX_SECRET_LABEL,
+            )
+            return secret
+
+    def publish_auth_data(
+        self,
+        auth_data: AuthDataConfig,
+    ):
+        """Publish litmus auth service data to the requirers."""
+        dex_secret = self._store_dex_config_in_secret(auth_data.dex_config)
+
+        for relation in self._relations:
+            try:
+                relation.save(
+                    AuthProviderAppDatabagModel(
+                        grpc_server_host=auth_data.grpc_server_host,
+                        grpc_server_port=auth_data.grpc_server_port,
+                        insecure=auth_data.insecure,
+                        dex_config=(
+                            {
+                                **auth_data.dex_config.model_dump(),
+                                **(
+                                    {"auth_secret_id": dex_secret.get_info().id}
+                                    if dex_secret
+                                    else {}
+                                ),
+                            }
+                        )
+                        if auth_data.dex_config
+                        else None,
+                    ),
+                    self._app,
+                )
+            except ops.ModelError:
+                logger.debug("failed to validate app data; is the relation still being created?")
+                continue
+
+            # grant the requirer access to the secret
+            if dex_secret:
+                dex_secret.grant(relation)
+
+    def get_requirer_endpoint(self) -> List[Endpoint]:
```

Comment:
```suggestion
    def get_backend_grpc_endpoints(self) -> List[Endpoint]:
```

This is not really a generic interface, but between auth and backend, so instead of keeping things generic and speak of providers/requirers, we should speak of authservers and backends, I think things will be more clear then